components/ActiveStream/index.tsx

```diff
@@ -0,0 +1,495 @@
+  const { switchNetwork, isLoading: switchIsLoading } = useSwitchNetwork();
+  const { data: reserve, isLoading: reserveIsLoading } = useEscrowedEXAReserves(BigInt(tokenId));
+  const { data: withdrawable, isLoading: withdrawableIsLoading } = useSablierV2LockupLinearWithdrawableAmountOf(
+    BigInt(tokenId),
+  );
+  const escrowedEXA = useEscrowedEXA();
+  const [loading, setLoading] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState<'nft' | 'cancel'>('nft');
+  const { breakpoints } = useTheme();
+  const isMobile = useMediaQuery(breakpoints.down('sm'));
+  const { data: nft } = useSablierV2NftDescriptorTokenUri(BigInt(tokenId));
+
+  const WithdrawAndCancel = () => {
+    return (
+      <>
+        <DialogTitle
+          sx={{
+            cursor: { xs: '', sm: 'move' },
+          }}
+        >
+          <Box
+            sx={{
+              padding: { xs: spacing(2, 1, 1), sm: spacing(1, 2, 0, 2) },
+            }}
+          >
+            <Typography fontWeight={700} fontSize={24}>
+              {t('Withdraw Reserved EXA')}
+            </Typography>
+          </Box>
+        </DialogTitle>
+        <Box
+          sx={{
+            padding: { xs: spacing(2, 1, 1), sm: spacing(2, 4, 4) },
+          }}
+        >
+          <DialogContent sx={{ p: 1, overflow: 'hidden' }}>
+            <Typography fontSize={14} fontWeight={500}>
+              {t(
+                'When you withdraw the reserved EXA, the associated vesting stream will be cancelled automatically. You’ll be able to claim the earned EXA and will get back all remaining esEXA.',
+              )}
+            </Typography>
+            <Box display="flex" gap={2} mt={4} alignItems="center">
+              {impersonateActive ? (
+                <Button fullWidth variant="contained">
+                  {t('Exit Read-Only Mode')}
+                </Button>
+              ) : chain && chain.id !== displayNetwork.id ? (
+                <LoadingButton
+                  fullWidth
+                  onClick={() => switchNetwork?.(displayNetwork.id)}
+                  variant="contained"
+                  loading={switchIsLoading}
+                >
+                  {t('Please switch to {{network}} network', { network: displayNetwork.name })}
+                </LoadingButton>
+              ) : (
+                <>
+                  <LoadingButton
+                    fullWidth
+                    variant="contained"
+                    color="error"
+                    onClick={handleCancel}
+                    loading={loading}
+                    data-testid={`vesting-stream-${tokenId}-cancel-submit`}
+                  >
+                    {t('Withdraw and Cancel Stream')}
+                  </LoadingButton>
+                </>
+              )}
+            </Box>
+          </DialogContent>
+        </Box>
+      </>
+    );
+  };
+
+  const NFT = () => {
```

Comment:
please, dont declare a component inside another component, as this will be re-created on each render, declare outside and pass props